.github/workflows/check.yaml

```diff
@@ -78,10 +78,12 @@ jobs:
         run: sudo snap install snapcraft --classic
       - name: Build snap
         run: snapcraft --use-lxd
+      - name: Determine system architecture
+        run: echo "SYSTEM_ARCH=$(uname -m)" >> $GITHUB_ENV
       - name: Upload the built snap as an artifact
         uses: actions/upload-artifact@v4
         with:
-          name: SNAP_FILE
+          name: snap_$${{ env.SYSTEM_ARCH }}
```

Comment:
Ah sorry, I didn't take a close enough look at this - there are copy/paste issues here: the double `$$` is still here from the terraform template.

Reply:
Fixed in https://github.com/canonical/charmed-openstack-upgrader/pull/569